enhancements/proposals/cleanup_after_scaling.md

```diff
@@ -0,0 +1,140 @@
+# Run node cleanup after scaling the cluster
+
+## Summary
+
+The Operator supports both horizontal and vertical scaling, but the procedure isn’t in sync with ScyllaDB documentation,
+because the cleanup part is not implemented. It’s important because upon scaling, the stored on node disk might become
+stale taking up unnecessary space. Operator should support running the cleanup to keep disk space as low as possible and
+ensure clusters are stable and reliable over time.
+
+## Motivation
+
+The motivation behind running node cleanup after scaling a cluster is to avoid the accumulation of unnecessary data on
+the node disks. When nodes are added or removed from the cluster, they gain or lose some tokens, which can result in
+files stored on the node disks still containing data associated with lost tokens. Over time, this can lead to a build-up
+of unnecessary data and cause disk space issues. By running node cleanup after scaling, these files can be cleared,
+freeing up disk space.
+
+### Goals
+
+* Running a node cleanup procedure after horizontally scaling the cluster.
+* Running a node cleanup procedure after scaling is completed, on all nodes but the last one added.
+
+### Non-Goals
+
+Running node cleanup during off-peak hours.
+
+## Proposal
+
+I propose to extend the sidecar with a new controller that will project the latest cleaned up and existing hash of
+tokens for each node as an annotation in the member service.
+In addition, a new controller in Operator will be responsible for managing Jobs that will execute a cleanup on nodes
+that require it. The trigger for the Job creation will be a mismatch between the current and latest hash. The controller
+will ensure that there will be only one cleanup Job running at the same time to prevent extraneous load on the cluster
```

Comment:
nit: s/Job/job/

Should be one cleanup job per node I guess.
But we can (and should) run cleanup on all nodes in parallel.

Reply:
fixed